lib/puppet/provider/hocon_setting/ruby.rb

```diff
@@ -82,8 +93,32 @@ def conf_object
     Hocon::ConfigFactory.parse_file(file_path)
   end
 
+  def remove_value(value_to_remove)
+    if resource[:type] == 'array_element'
+      new_value_tmp = []
+      val = value
+      new_value_tmp << val
+      new_value_tmp.flatten!
+      Array(value_to_remove).each do |v|
+        new_value_tmp.delete(v)
+      end
+      new_value = Hocon::ConfigValueFactory.from_any_ref(new_value_tmp, nil)
+      conf_file_modified = conf_file.set_config_value(setting, new_value)
+    else
+      conf_file_modified = conf_file.remove_value(setting)
+    end
+    conf_file_modified
+  end
+
   def set_value(value_to_set)
-    if resource[:type] == 'array' || (value_to_set.is_a?(String) && resource[:type] != 'text') || (value_to_set.is_a?(Array) && value_to_set.size > 1)
+    if resource[:type] == 'array_element'
+      tmp_val = []
+      tmp_val << value
+      tmp_val << value_to_set
+      tmp_val.flatten!
```

Comment:
My concern with not having `uniq!` for the `array_element` is idempotence. Since the array can't be  rewritten every time we'd just end up with duplicate elements each run. I think if I iterate through `value` and `value_to_set` I can get rid of `flatten!` though.